src/core/linalg/src/sparse/4C_linalg_sparsematrix.cpp

```diff
@@ -1631,47 +1589,47 @@ int Core::LinAlg::SparseMatrix::extract_global_row_view(
 /*----------------------------------------------------------------------*
  *----------------------------------------------------------------------*/
 int Core::LinAlg::SparseMatrix::insert_my_values(
-    int my_row, int num_entries, const double* values, const int* indices) const
+    int my_row, int num_entries, const double* values, const int* indices)
 {
   return sysmat_->InsertMyValues(my_row, num_entries, values, indices);
 }
 
 /*----------------------------------------------------------------------*
  *----------------------------------------------------------------------*/
 int Core::LinAlg::SparseMatrix::sum_into_my_values(
-    int my_row, int num_entries, const double* values, const int* indices) const
+    int my_row, int num_entries, const double* values, const int* indices)
 {
   return sysmat_->SumIntoMyValues(my_row, num_entries, values, indices);
 }
 
 /*----------------------------------------------------------------------*
  *----------------------------------------------------------------------*/
 int Core::LinAlg::SparseMatrix::replace_my_values(
-    int my_row, int num_entries, const double* values, const int* indices) const
+    int my_row, int num_entries, const double* values, const int* indices)
 {
   return sysmat_->ReplaceMyValues(my_row, num_entries, values, indices);
 }
 
 /*----------------------------------------------------------------------*
  *----------------------------------------------------------------------*/
 int Core::LinAlg::SparseMatrix::replace_global_values(
-    int global_row, int num_entries, const double* values, const int* indices) const
+    int global_row, int num_entries, const double* values, const int* indices)
 {
   return sysmat_->ReplaceGlobalValues(global_row, num_entries, values, indices);
 }
 
 /*----------------------------------------------------------------------*
  *----------------------------------------------------------------------*/
 int Core::LinAlg::SparseMatrix::insert_global_values(
-    int global_row, int num_entries, const double* values, const int* indices) const
+    int global_row, int num_entries, const double* values, const int* indices)
 {
   return sysmat_->InsertGlobalValues(global_row, num_entries, values, indices);
```

Comment:
The constant `const` violations in our wrappers prompted me to create `OwnerOrView` (https://github.com/4C-multiphysics/4C/blob/main/src/core/utils/src/stl_extension/4C_utils_owner_or_view.hpp) which wraps underlying objects and propagates const-ness. It might be interesting to apply it here as well.